src/main/webapp/courseLoader.js

```diff
@@ -2,126 +2,216 @@ google.charts.load("current", { packages: ["corechart"] });
+  );
+  const termPerceptionOptions = {
+    title: "Perception of Term Reviews",
+    legend: { position: "none" },
+    vAxis: { title: "Perception" },
+    hAxis: { title: "Comment Quantity" },
+    histogram: {
+      hideBucketItems: true,
+    },
+  };
+  const termPerceptionChart = new google.visualization.Histogram(
+    document.getElementById("term-chart")
+  );
+  termPerceptionChart.draw(termPerceptionData, termPerceptionOptions);
+
+  const tempProfPerceptionList = dataObject.termPerceptionList;
+  const profPerceptionList = [
+    ["Professor Perception"],
+    /* dummyPerceptionRating */ [2],
+    /* dummyPerceptionRating */ [9],
+  ].concat(tempProfPerceptionList);
+  const profPerceptionData = new google.visualization.arrayToDataTable(
+    profPerceptionList
+  );
+  const profPerceptionOptions = {
+    title: "Perception of Professor Reviews",
+    legend: { position: "none" },
+    vAxis: { title: "Perception" },
+    hAxis: { title: "Comment Quantity" },
+    histogram: {
+      hideBucketItems: true,
+    },
+  };
+  const profPerceptionChart = new google.visualization.Histogram(
+    document.getElementById("prof-chart")
+  );
+  profPerceptionChart.draw(profPerceptionData, profPerceptionOptions);
+}
+
+async function postData(url, data = {}) {
+  // Default options are marked with *
+  const response = await fetch(url, {
+    method: "POST", // *GET, POST, PUT, DELETE, etc.
+    cache: "no-cache",
+    credentials: "same-origin", // include, *same-origin, omit
+    headers: {
+      "Content-Type": "application/json",
+      // 'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: JSON.stringify(data), // body data type must match "Content-Type" header
+  });
+  return response.json(); // parses JSON response into native JavaScript objects
+}
+
+function storeData() {
+  var data = {};
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const courseName = urlParams.get("course-name");
+  const term = urlParams.get("term");
+  const profName = urlParams.get("prof-name");
+  const units = urlParams.get("num-units");
+  const schoolName = urlParams.get("school-name");
+
+  data["course-name"] = courseName;
+  data["term"] = term;
+  data["prof-name"] = profName;
+  data["units"] = units;
+  data["school-name"] = schoolName;
+
+  const termInput = document.getElementById("term-input").value;
+  const profInput = document.getElementById("prof-input").value;
+  const ratingTerm = document.getElementById("rating-term").value;
+  const ratingProf = document.getElementById("rating-prof").value;
+  const hours = document.getElementById("hoursOfWork").value;
+  const diff = document.getElementById("difficulty").value;
+  document.getElementById("termForm").reset();
+
+  data["term-input"] = termInput;
+  data["prof-input"] = profInput;
+  data["rating-term"] = ratingTerm;
+  data["rating-prof"] = ratingProf;
+  data["hours"] = hours;
+  data["difficulty"] = diff;
```

Comment:
I'd use the shorthand object creation here instead.

So this would turn into:
```
const data = {
   "course-name": urlParams.get("course-name"),
   ...
   "term-input": document.getElementById("term-input").value,
}
```

Little easier to read

If you use camelCase, then when constructing the new url pass data.schoolName etc instead.

Reply:
Also, you don't need the "" if you're not using dash-case.